api/types.go

```diff
@@ -88,8 +104,16 @@ type LocationProfile struct {
 	RestrictAltLoc     bool     `json:"restrict_alt_loc,omitempty"`
 }
 
-// @TODO(mohmann): add necessary fields
-type ThresholdProfile struct{}
+// ThresholdProfile help the alarms engine to decide if a specific resource has to be declared critical or down
+type ThresholdProfile struct {
+	ProfileID              string              `json:"profile_id"`
+	Type                   string              `json:"type"`
+	ProfileName            string              `json:"profile_name"`
+	DownLocationThreshold  int                 `json:"down_location_threshold"`
+	WebsiteContentModified bool                `json:"website_content_modified,omitempty"`
+	WebsiteContentChanges  []ContentChange     `json:website_content_changes,omitempty"`
+	ResponseTimeThreshold  []ResponseThreshold `json:"response_time_threshold,omitempty"`
```

Comment:
Change the type from `[]ResponseThreshold` to `interface{}` to support arbitrary structured json objects.